test/unit/Chainweb/Test/Pact4/PactExec.hs

```diff
@@ -572,14 +576,19 @@ execLocalTest runPact name (trans',check) = testCase name (go >>= check)
       results' <- tryAny $ runPact $
         execLocal (Pact4.unparseTransaction trans) Nothing Nothing Nothing
       case results' of
-        Right (MetadataValidationFailure e) ->
+        Right (preview _MetadataValidationFailure -> Just e) ->
+          return $ Left $ show e
+        Right (preview _LocalTimeout -> Just ()) ->
+          return $ Left "LocalTimeout"
+        Right (preview _LocalResultLegacy -> Just cr) -> do
+          Just decodedCr <- return $ Aeson.decode (TL.encodeUtf8 $ TL.fromStrict $ J.getJsonText cr)
+          return $ Right decodedCr
+        Right (preview _LocalResultWithWarns -> Just (cr, _)) -> do
+          Just decodedCr <- return $ Aeson.decode (TL.encodeUtf8 $ TL.fromStrict $ J.getJsonText cr)
+          return $ Right decodedCr
```

Comment:
Slight bike shed because it's test code so it's not necessary, but you could factor out the repetition after the pattern match.

Reply:
I'll probably just delete this test code after Pact 4 is mostly out.